### ISSUE TITLE
Artificial gaps when plotting discrete data in distplot

### ISSUE THREAD

**Issue by fonnesbeck** [mwaskom/seaborn#327]
An issue that arises when plotting histograms of discrete data with Matplotlib seems to also come up in `distplot` as well. That is, there are often empty bins in the histogram that are entirely artifacts of the plot. See here for an example with Poisson data:

![poisson data](http://d.pr/i/1e28U+)

Is there any way that `distplot` could try to detect this and adjust the bins automagically?

**Comment by mwaskom** [comment]
Twitter conversation made me remember this.

Last I thought about it, I wasn't sure that changing behavior depending on whether the input data were int or float was the best way to go. a) I can't think of anywhere else that is important in seaborn, so it's less predictable. b), Numeric pandas data often end up as object-typed for one reason or another (also this is what you have to do if you want integers with missing data, right?) So figuring out how to define "integer data" might be tricky.

Perhaps a different option would be to make it easier to define the bins in this case? Maybe let `bins` take a `(min, max, n)` tuple that gets passed to `np.linspace` to save some typing? Or `(min, max, [n])` where `n` is a sensible default (probably the default reference rule, so that wouldn't help you with discrete data).

**Comment by johannesreiter** [comment]
I have exactly the same problem. The problem is certainly not that some bins are zero. The placement of the bars seems to be non-deterministic.
<img width="1018" alt="screen shot 2016-08-10 at 4 07 23 pm" src="https://cloud.githubusercontent.com/assets/6780928/17569083/b353c73a-5f14-11e6-9157-00b04eb50799.png">

**Comment by mwaskom** [comment]
yes you need to do `bins=np.arange(data.min(), data.max() + 1)`

**Comment by mwaskom** [comment]
Closed with #2125:

```python
x = np.random.poisson(10, size=1000)
sns.histplot(x, discrete=True)
```
![image](https://user-images.githubusercontent.com/315810/84604377-8edc0280-ae63-11ea-8641-0657a9dfacb7.png)

**Comment by fonnesbeck** [comment]
Thanks @mwaskom!
